### ISSUE TITLE
Should I upgrade? Will nothing break?

### ISSUE THREAD

**Issue by kbrmimbyl** [flutter/flutter#28818]
I have a fully working flutter installation and I'm about to take my app to production. I'm working on 2 dev environments -- Ubuntu 16.04 and MacOS High Sierra.

Here's the output of `flutter --version` on both:

```
Flutter 1.0.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 5391447fae (3 months ago) • 2018-11-29 19:41:26 -0800
Engine • revision 7375a0f414
Tools • Dart 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)
```

From the last couple of days I'm being asked to upgrade flutter. I'm very hesitant since I'm close to production... what if my app breaks? Can someone help me with the following questions?

1. Is my concern valid? If yes, should I just refrain from upgrading?
2. What's the best way to test the latest version and revert to 1.0.0 if my app breaks? Note: I'm using specific plugin versions for most plugins and I'm not willing to upgrade any plugin right now.
3. What's the latest version to which I'm being asked to upgrade (neither flutter nor the VSCode extension report it) ?
4. Where are the release notes from which I can tell if things will break?

Please help!

**Comment by jonahwilliams** [comment]
Hi @kbrmimbyl , if you have concerns about breaking changes you can check the [changelog](https://github.com/flutter/flutter/wiki/Changelog), specifically in the "Changes since v1.0.0" section.

The latest stable release of flutter is version 1.2. If you have problems with upgrading then you can always revert to 1.0.0 by checking out the 1.0.0 tag or running `flutter version -f 1.0.0.

Regarding plugins, I do not know which ones you are using, so I cannot say if you will have issues. You should be aware of the AndroidX compatibility issues if upgrading plugins.

**Comment by jonahwilliams** [comment]
Hopefully I've answered your questions - I'm going to close this issue for now, thanks!

**Comment by kbrmimbyl** [comment]
Yes, thanks a lot for your help, @jonahwilliams !

**Comment by Wuzo** [comment]
> Hi @kbrmimbyl , if you have concerns about breaking changes you can check the [changelog](https://github.com/flutter/flutter/wiki/Changelog), specifically in the "Changes since v1.0.0" section.
> 
> The latest stable release of flutter is version 1.2. If you have problems with upgrading then you can always revert to 1.0.0 by checking out the 1.0.0 tag or running `flutter version -f 1.0.0.
> 
> Regarding plugins, I do not know which ones you are using, so I cannot say if you will have issues. You should be aware of the AndroidX compatibility issues if upgrading plugins.

"flutter version -f 1.0.0 "
Could not find a command named "version".

**Comment by yathit** [comment]
@Wuzo flutter version is only available in 1.2.

**Comment by Wuzo** [comment]
> @Wuzo flutter版仅适用于1.2。

thx
